scan_tutorial/scan_tutorial.ipynb

```diff
@@ -0,0 +1,735 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Scan\n",
+    "\n",
+    "## In short\n",
+    "\n",
+    "* Mechanism to perform loops in a Theano graph\n",
+    "* Supports nested loops and reusing results from previous iterations \n",
+    "* Highly generic\n",
+    "\n",
+    "## Implementation\n",
+    "\n",
+    "You've previous seen that a Theano function graph is composed of two types of nodes; Variable nodes which represent data and Apply node which apply Ops (which represent some computation) to Variables to produce new Variables.\n",
```

Comment:
Right. I'll still introduce the concept the first day, especially since I demonstrate pydotprint as well.